From 48c575f9d5127f7c6c7548f9be3b48e93a1efffe Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:38 -0400
Subject: [ethernet] i40e: Do not expose fd-sb commands from debugfs

Message-id: <1409658042-22689-98-git-send-email-sassmann@redhat.com>
Patchwork-id: 90291
O-Subject: [RHEL7.1 PATCH 097/281] i40e: Do not expose fd-sb commands from debugfs
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

Ethtool -k/-K can handle this so we do not need it in debugfs.

Change-ID: I9df692f10a60b71805f8f48d3b87c7da3820b2aa
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 53275f28f84b91c6c581242f8fec23f7104d69f0)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
index 3c37386..1aaec40 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
@@ -1744,10 +1744,6 @@ static ssize_t i40e_dbg_command_write(struct file *filp,
   i40e_dbg_cmd_fd_ctrl(pf, I40E_FLAG_FD_ATR_ENABLED, false);
  } else if (strncmp(cmd_buf, "fd-atr on", 9) == 0) {
   i40e_dbg_cmd_fd_ctrl(pf, I40E_FLAG_FD_ATR_ENABLED, true);
- } else if (strncmp(cmd_buf, "fd-sb off", 9) == 0) {
-  i40e_dbg_cmd_fd_ctrl(pf, I40E_FLAG_FD_SB_ENABLED, false);
- } else if (strncmp(cmd_buf, "fd-sb on", 8) == 0) {
-  i40e_dbg_cmd_fd_ctrl(pf, I40E_FLAG_FD_SB_ENABLED, true);
  } else if (strncmp(cmd_buf, "lldp", 4) == 0) {
   if (strncmp(&cmd_buf[5], "stop", 4) == 0) {
    int ret;
@@ -1967,8 +1963,6 @@ static ssize_t i40e_dbg_command_write(struct file *filp,
   dev_info(&pf->pdev->dev, "  rem fd_filter <dest q_index> <flex_off> <pctype> <dest_vsi> <dest_ctl> <fd_status> <cnt_index> <fd_id> <packet_len> <packet>\n");
   dev_info(&pf->pdev->dev, "  fd-atr off\n");
   dev_info(&pf->pdev->dev, "  fd-atr on\n");
-  dev_info(&pf->pdev->dev, "  fd-sb off\n");
-  dev_info(&pf->pdev->dev, "  fd-sb on\n");
   dev_info(&pf->pdev->dev, "  lldp start\n");
   dev_info(&pf->pdev->dev, "  lldp stop\n");
   dev_info(&pf->pdev->dev, "  lldp get local\n");
-- 
1.7.1